hera_qm/xrfi.py

```diff
@@ -1639,6 +1658,310 @@ def xrfi_run(ocalfits_file, acalfits_file, model_file, data_file, history,
+                                           uvf_v2, uvf_d2, uvf_chisq2],
+                                          method='quadmean', inplace=False)
+    uvf_metrics2.label = 'Combined metrics, round 2.'
+    alg_func = algorithm_dict['detrend_meanfilt']
+    uvf_metrics2.metric_array[:, :, 0] = alg_func(uvf_metrics2.metric_array[:, :, 0],
+                                                  flags=uvf_init.flag_array[:, :, 0],
+                                                  Kt=kt_size, Kf=kf_size)
+
+    # Flag on combined metrics
+    uvf_f2 = flag(uvf_metrics2, nsig_p=sig_init, run_check=run_check,
+                  check_extra=check_extra,
+                  run_check_acceptability=run_check_acceptability)
+    uvf_fws2 = watershed_flag(uvf_metrics2, uvf_f2, nsig_p=sig_adj,
+                              inplace=False, run_check=run_check,
+                              check_extra=check_extra,
+                              run_check_acceptability=run_check_acceptability)
+    uvf_fws2.label = 'Flags from combined metrics, round 2.'
+    uvf_combined2 = (uvf_fws2 | uvf_ogf2 | uvf_oxf2 | uvf_agf2 | uvf_axf2
+                     | uvf_vf2 | uvf_df2 | uvf_chisq_f2 | uvf_init)
+    uvf_combined2.label = 'ORd flags, round 2.'
+
+    # Write everything out
+    uvf_dict = {'apriori_flags.h5': uvf_apriori,
+                'v_metrics1.h5': uvf_v, 'v_flags1.h5': uvf_vf,
+                'og_metrics1.h5': uvf_og, 'og_flags1.h5': uvf_ogf,
+                'ox_metrics1.h5': uvf_ox, 'ox_flags1.h5': uvf_oxf,
+                'ag_metrics1.h5': uvf_ag, 'ag_flags1.h5': uvf_agf,
+                'ax_metrics1.h5': uvf_ax, 'ax_flags1.h5': uvf_axf,
+                'chi_sq_renormed1.h5': uvf_chisq, 'chi_sq_flags1.h5': uvf_chisq_f,
+                'combined_metrics1.h5': uvf_metrics, 'combined_flags1.h5': uvf_fws,
+                'flags1.h5': uvf_init,
+                'v_metrics2.h5': uvf_v2, 'v_flags2.h5': uvf_vf2,
+                'og_metrics2.h5': uvf_og2, 'og_flags2.h5': uvf_ogf2,
+                'ox_metrics2.h5': uvf_ox2, 'ox_flags2.h5': uvf_oxf2,
+                'ag_metrics2.h5': uvf_ag2, 'ag_flags2.h5': uvf_agf2,
+                'ax_metrics2.h5': uvf_ax2, 'ax_flags2.h5': uvf_axf2,
+                'data_metrics2.h5': uvf_d2, 'data_flags2.h5': uvf_df2,
+                'chi_sq_renormed2.h5': uvf_chisq2, 'chi_sq_flags2.h5': uvf_chisq_f2,
+                'combined_metrics2.h5': uvf_metrics2, 'combined_flags2.h5': uvf_fws2,
+                'flags2.h5': uvf_combined2}
+
+    # Determine the actual files to store
+    uvtemp = UVData()
+    uvtemp.read(data_files[0], read_data=False)
+    nintegrations = len(data_files) * uvtemp.Ntimes
+    ndrop = int(np.ceil(kt_size / uvtemp.Ntimes))
+    start_ind = ndrop
+    end_ind = len(data_files) - ndrop
+    # If we're the first or last job, store all flags for the edge
+    datadir = os.path.dirname(os.path.abspath(data_files[0]))
+    all_files = sorted(glob.glob(os.path.join(datadir, '*sum.uvh5')))
```

Comment:
Not wild about hardcoding this...

Reply:
Can we have it figure out the files based on the data files passed in.

Reply:
I don't think so. This is just used to determine the edges of the entire night, which is usually outside the files that are passed in. I've updated the search to be more robust.